test_lbaas.py

```diff
@@ -0,0 +1,625 @@
+    * Send TCP traffic through the LB and verify backend answers.
+
+    """
+
+    # Create a load balancer setup with two backends on a private network
+    load_balancer, listener, pool, backends, private_network = \
+        create_load_balancer_scenario()
+
+    # Issue 10 requests on IPv4 and IPv6 to the load balancer
+    for i in range(10):
+        content = load_balancer.get_url(prober, addr_family=4)
+        assert 'Backend server running on' in content
+
+        content = load_balancer.get_url(prober, addr_family=6)
+        assert 'Backend server running on' in content
+
+    # Assert logs on both backend servers show they received traffic
+    assert set(backends) == set(get_backends_for_request(backends))
+
+
+def test_multi_listener(prober, create_load_balancer_scenario):
+    """ Two load balancer listeners connected to the same pool.
+
+    """
+
+    # Create a load balancer setup with one backends on a private network
+    load_balancer, listener1, pool, (backend, ), private_network = \
+        create_load_balancer_scenario(num_backends=1)
+
+    # Add an additional listener on Port 81
+    load_balancer.add_listener('listener-81', pool, 81)
+
+    # Assert the LB works on port 80
+    assert (prober.http_get(f'http://{load_balancer.vip(4)}/hostname')
+            == backend.name)
+
+    # Assert backend is also reachable on port 81 (try for 20s as it might not
+    # be ready yet)
+    retry_for(seconds=30).or_fail(
+        load_balancer.verify_backend,
+        msg='Additional listener not reachable within 30s.',
+        prober=prober,
+        backend=backend,
+        port=81,
+    )
+
+
+def test_multi_listener_multi_pool(prober, create_server, image,
+                                   create_private_network,
+                                   create_load_balancer_scenario):
+    """ Two listeners connected to their own pool of member servers each.
+
+    """
+
+    # Create a load balancer setup with one backends on a private network
+    load_balancer, listener1, pool1, (backend1, ), private_network1 = \
+        create_load_balancer_scenario(num_backends=1)
+
+    # Create an additonal backend server
+    backend2 = create_server(
+        name='backend2',
+        image=image,
+        use_public_network=False,
+        use_private_network=True,
+        jump_host=prober,
+    )
+
+    # Create a second pool
+    pool2 = load_balancer.add_pool(f'lb-pool-2', 'round_robin', 'tcp')
+
+    # create an additional backend network
+    private_network2 = create_private_network(auto_create_ipv4_subnet=True)
+
+    # Create an additional backend server for the second pool
+    setup_lbaas_backend(backend2, load_balancer, pool2, private_network2)
+
+    # Add an additional listener on Port 81 for the second pool
+    load_balancer.add_listener('listener-81', pool2, 81)
+
+    # Assert backend1 is reachable on port 80 (must already be ready)
+    assert (prober.http_get(f'http://{load_balancer.vip(4)}/hostname')
```

Comment:
I guess you meant the parenthesis. I first had `\` line continuation but the changed to this style because at least some coding styles I know strongly discourage `\`. I changed it back now in a separate commit, so I can easily revert it if someone else has another opinion on this. I personally don't care much.